clients/src/test/java/org/apache/kafka/common/utils/ShellTest.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.common.utils;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.Timeout;
+
+import java.util.Locale;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assume.assumeTrue;
+
+public class ShellTest {
+    @Rule
+    public final Timeout globalTimeout = Timeout.seconds(180);
+
+    @Test
+    public void testEchoHello() throws Exception {
+        assumeTrue(!System.getProperty("os.name").toLowerCase(Locale.ROOT).startsWith("windows"));
```

Comment:
Sorry, this is a bit annoying, but we have a simple class `Os` in the Scala code that has this logic. Can we convert it to Java, move it to `org.apache.commons.internals` and use it here?